mne/viz/_mpl_figure.py

```diff
@@ -2318,6 +2330,150 @@ def _get_scale_bar_texts(self):
+        time_start = event.time_start
+        time_end = event.time_end
+
+        if self.mne.is_epochs:
+            # Translate the start/end times from the original raw to the indices of the
+            # epochs being shown, and then to the appropriate start/end times in the
+            # browser.
+            events = self.mne.inst.events[self.mne.inst.selection]
+            onsets = events[:, 0] / self.mne.info["sfreq"]
+            # Subtract/add one sample to make sure we end on the right side. This is
+            # needed because of small floating point errors.
+            epoch_ix_start = np.searchsorted(
+                onsets, time_start - self.mne.inst.tmin - self.mne.sampling_period
+            )
+            epoch_ix_end = np.searchsorted(
+                onsets, time_end - self.mne.inst.tmax + self.mne.sampling_period
+            )
+            # Always show at least one epoch.
+            epoch_ix_start = min(epoch_ix_start, len(self.mne.inst) - 1)
+            epoch_ix_end = max(epoch_ix_start + 1, epoch_ix_end)
+            self.mne.n_epochs = epoch_ix_end - epoch_ix_start
+
+            # Compute the browser time period to match the selected epochs.
+            time_start = self.mne.boundary_times[epoch_ix_start]
+            width = self.mne.boundary_times[1] - self.mne.boundary_times[0]
+            time_end = self.mne.boundary_times[epoch_ix_end - 1] + width
+        else:
+            # For raw data, we need to take `first_time` into account.
+            time_start -= self.mne.inst.first_time
+            time_end -= self.mne.inst.first_time
+
+            # Never show fewer than 3 samples.
+            min_dur = 3 * np.diff(self.mne.inst.times[:2])[0]
+            time_end = np.clip(time_end, time_start + min_dur, self.mne.inst.times[-1])
+
+        # Update browser window.
+        with disable_ui_events(self):
+            self.mne.t_start = time_start
+            self.mne.duration = time_end - time_start
+            self._update_hscroll()
+            self._redraw(annotations=True)
+
+    # def _on_channels_select_event(self, event):
+    #     """Respond to the ChannelsSelect UI event."""
+    #     old_n_channels = self.mne.n_channels
+    #     picks = np.flatnonzero(
+    #         np.isin(self.mne.ch_names[self.mne.ch_order], event.ch_names)
+    #     )
+    #     if len(picks) == 0:
+    #         return  # can't handle the event
+    #     if picks.min() == self.mne.ch_start and len(picks) == self.mne.n_channels:
+    #         return  # no change
+
+    #     with disable_ui_events(self):
+    #         self.mne.ch_start = picks.min()
+    #         self.mne.n_channels = len(picks)
+    #         self._update_vscroll()
+    #         self._update_picks()
+    #         if self.mne.n_channels != old_n_channels:
+    #             self._update_trace_offsets()
+    #         self._redraw(annotations=True)
+
+    def _on_channels_select_event(self, event):
```

Comment:
As far as I can tell, this "new" function does exactly the same as the one commented-out above, minus the `disable_ui_events`. Guess this is a work in progress thing.